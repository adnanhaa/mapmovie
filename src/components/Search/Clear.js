import style from './Clear.css';
import React, {useCallback} from "react";
import * as PropTypes from "prop-types";

/*
* local component - only for Search
* TODO create function after review
*/
const Clear = ({text, clickHandler}) => {

    const handleClick = useCallback(() => {
        clickHandler();
    }, []);

    return <div
        className={style.Clear}
        onClick={handleClick}>
        {text}
    </div>
};

//TODO improve this
Clear.defaultProps = { text: 'clear' };
Clear.propTypes = {
    text : PropTypes.string,
    clickHandler : PropTypes.func
};

export default Clear;
