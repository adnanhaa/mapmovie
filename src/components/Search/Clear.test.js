import React from 'react';
import {shallow} from 'enzyme';
import Clear from "./Clear";


describe('Clear', () => {
    it('should render correctly in "debug" mode', () => {
        const component = shallow(<Clear debug />);
        expect(component).toMatchSnapshot();
    });

    it('should render correctly with no props', () => {
        const component = shallow(<Clear/>);
        expect(component).toMatchSnapshot();
    });

    it('should render banner text correctly with given strings', () => {
        const string = 'clear';
        const component = shallow(<Clear text={string} />);
        expect(component).toMatchSnapshot();
    });

});