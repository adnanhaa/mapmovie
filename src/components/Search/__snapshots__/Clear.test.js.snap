// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clear should render banner text correctly with given strings 1`] = `
<div
  onClick={[Function]}
>
  clear
</div>
`;

exports[`Clear should render correctly in "debug" mode 1`] = `
<div
  onClick={[Function]}
>
  clear
</div>
`;

exports[`Clear should render correctly with no props 1`] = `
<div
  onClick={[Function]}
>
  clear
</div>
`;
